src/connection_string.ts

```diff
@@ -369,6 +368,8 @@ export function parseOptions(
     }
   }
 
+  checkTLSOptions(allOptions);
```

Comment:
Update check from `mongoOptions` to `allOptions` as `mongoOptions` object does not have key value pairs [when value would be falsy](https://github.com/mongodb/node-mongodb-native/blob/8b0ba59b4142237adaf19fa5bbe8c6195d2e34a0/src/connection_string.ts#L389).